File: apps/api-dotnet/src/Planora.Application/Features/Notifications/Dtos/NotificationDto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Planora.Domain.Enums;

namespace Planora.Application.Features.Notifications.Dtos
{
    public record NotificationDto(
        Guid Id,
        NotificationType Type,
        string Title,
        string Message,
        string? Link,
        string? Data,
        DateTime CreatedAt,
        bool IsRead);
}

File: apps/frontend/src/app/core/interfaces/notification/notification-dto.ts
import { NotificationType } from "./notification-type";

export interface NotificationDto {
  id: string;
  type: NotificationType;
  title: string;
  message: string;
  link: string | null;
  data: string | null;
  createdAt: string;
  isRead: boolean;
}
